chore: fix warnings

sublimator opened the pull request:
```
git checkout origin/tmp-conan -- \
    src/test/core/SociDB_test.cpp \
    src/ripple/app/rdb/backend/RWDBDatabase.h \
    src/ripple/overlay/impl/Handshake.cpp
```

sublimator committed:
chore: fix warnings

dangell7 commented on src/ripple/overlay/impl/Handshake.cpp:
```diff
@@ -239,8 +239,7 @@ verifyHandshake(
 
-    // Check the network. Omitting Network-ID (on either side ours, or theirs)
-    // means NID=0
+    if (auto const iter = headers.find("Network-ID"); iter != headers.end())
```
This doesn't seem right. Its synced code but it might have been merged correctly. These 2 calls are redundant

dangell7 committed:
fix: bad merge

dangell7 commented:
Changes in Handshake.cpp need an extra set of eyes. That was a bad merge. This is now exactly whats in ripple(d) but still want to make sure were not breaking anything.

sublimator commented on src/ripple/overlay/impl/Handshake.cpp:
```diff
@@ -241,18 +241,12 @@ verifyHandshake(
-                    peer_nid, std::string(iter->value())))
-                throw std::runtime_error("Invalid peer network identifier");
-        }
+        std::uint32_t nid;
```
ooof

RichardAH replied:
you gonna add a {0} here? (defensively) then we can merge?

sublimator replied:
@dangell7 ? 

I think this was something from the Conan branch 
Looks like you took over this PR

RichardAH replied:
I just updated it to current dev branch, do we need this pr or not?

dangell7 replied:
@RichardAH should be ready.

dangell7 committed:
fix build error

dangell7 committed:
add include

RichardAH committed:
Merge branch 'dev' into nd-fix-warnings-2025-05-06

RichardAH committed:
Merge branch 'dev' into nd-fix-warnings-2025-05-06

dangell7 committed:
add defensive 0

RichardAH commented on src/ripple/overlay/impl/Handshake.cpp:
```diff
@@ -239,19 +239,14 @@ verifyHandshake(
 
-        uint32_t our_nid = networkID ? *networkID : 0;
-        if (peer_nid != our_nid)
+        if (networkID && nid != *networkID)
```
this code has a different meaning than what it replaces... should be `if (!networkID && nid != 0 || networkID && nid != *networkID)` shouldn't it?

dangell7 replied:
Yes, your right. Updated

If peer sends no Network-ID header but local networkID is set: PR version passes, Suggestion version fails
If peer sends Network-ID but local networkID is unset: PR version passes, Suggestion version fails

dangell7 committed:
Tighten network ID check in handshake verification

dangell7 committed:
Merge branch 'dev' into nd-fix-warnings-2025-05-06

sublimator committed:
Merge remote-tracking branch 'origin/dev' into nd-fix-warnings-2025-05-06

sublimator committed:
simplify: network id verification logic in handshake

Use value_or(0) to treat absent/empty network ID as default network 0,
replacing verbose condition with equivalent simpler logic.

sublimator committed:
fix: network id check to handle missing header

Move network ID comparison outside the header check to ensure
rejection when local has networkID but peer doesn't send header
(implicitly network 0).

sublimator committed:
style: wrap network id check in scope block

Keep nid variable scoped and move comment to top of block
for better code organization.

sublimator commented:
We don't even /really/ need this any more, do we?!
It's a relic from a time when the conan branch had stricter warnings enabled, and we wanted to separate some changes in a separate PR iirc

RichardAH committed:
Merge branch 'dev' into nd-fix-warnings-2025-05-06
